lib/jekyll-mentions.rb

```diff
@@ -38,10 +38,7 @@ def filter_with_mention(src)
       end
 
       def mention_username_pattern
-        Regexp.new(
-          HTML::Pipeline::MentionFilter::UsernamePattern.source,
-          Regexp::IGNORECASE
-        )
+        @mention_username_pattern ||= %r![\w][\w_-]*!
```

Comment:
Is there a reason to change this here rather than in **HTML::Pipeline**?

Reply:
Does GitHub allow underscores in the username..? I couldn't find any documentation that said so..
Besides, its better to have a custom regex (that allows using valid twitter handles) here in the wrapper plugin instead of proposing a change in the main engine (`HTML::Pipeline`) which may not be open to support external usernames.

Reply:
> The value can be overriden by providing the username_pattern variable in the context.

Should we go this route instead?

This RegEx has the potential to be _very_ inefficient. If we are committing ourselves to bring this complexity into this plugin, we should see if we need to tighten it up at all.

Reply:
> Should we go this route instead?

That's how it is already.. This regex is passed to the context in html-profiler via `:filter_with_mention` method

> This RegEx has the potential to be very inefficient.

Really? I wonder and would definitely like to know how different this regex is in comparison to existing regex..

Reply:
> Really? I wonder and would definitely like to know how different this regex is in comparison to existing regex..

Sorry, I didn't mean that as a criticism; I see that you've mostly copied the existing regex.

I'm just worried that this regex isn't limiting itself to word boundaries or anything. I'm assuming it's only trying to match against the characters between `@` and the next space, but I haven't dug far enough into the code over there to figure it out.

I hope [this](https://github.com/jch/html-pipeline/blob/cdb943678efc905ec542487f33413e0ba0d322f3/lib/html/pipeline/%40mention_filter.rb#L107-L116) isn't looping through each word in every `text` node and trying to match that word with a `@` in front of it...

Reply:
I wonder if it would be worth it to add a negative lookbehind assertion, to make sure we are only trying to match words that follow `@`?

```ruby
@mention_username_pattern ||= %r!(?<=@)[\w][\w_-]*!
```

Reply:
> I hope this isn't looping through each word in every `text` node and trying to match

I think `:gsub` with a `regex` pattern does that anyways..

> I wonder if it would be worth it to add a negative lookbehind assertion

This is not necessary as the logic in `html-profiler` already includes a [regex with an atomic group](https://github.com/jch/html-pipeline/blob/cdb943678efc905ec542487f33413e0ba0d322f3/lib/html/pipeline/%40mention_filter.rb#L39-L52)..

```ruby
      # Hash that contains all of the mention patterns used by the pipeline
      MentionPatterns = Hash.new do |hash, key|
        hash[key] = /
          (?:^|\W)                    # beginning of string or non-word char
          @((?>#{key}))  # @username
          (?!\/)                      # without a trailing slash
          (?=
            \.+[ \t\W]|               # dots followed by space or non-word character
            \.+$|                     # dots at end of line
            [^0-9a-zA-Z_.]|           # non-word character except dot
            $                         # end of line
          )
        /ix
      end
```